File: src/main/java/me/delivery/domain/address/model/vo/AddressVO.java
package me.delivery.domain.address.model.vo;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import me.delivery.domain.address.model.entity.Address;
import org.springframework.data.geo.Point;

import java.math.BigDecimal;

@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class AddressVO {
    private long id;
    private String address;
    private String metropolitan;
    private String city;
    private String town;
    private String road;
    private Integer mainNumber;
    private Integer subNumber;
    private Point point;

    public static AddressVO from(Address address) {
        return AddressVO.builder()
                .id(address.getId())
                .address(address.getAddress())
                .metropolitan(address.getMetropolitan())
                .city(address.getCity())
                .town(address.getTown())
                .road(address.getRoad())
                .mainNumber(address.getMainNumber())
                .subNumber(address.getSubNumber())
                .point(address.getPoint())
                .build();
    }
}

File: src/main/kotlin/me/delivery/domain/address/model/vo/AddressVO.kt
package me.delivery.domain.address.model.vo

import me.delivery.domain.address.model.entity.Address
import org.springframework.data.geo.Point

class AddressVO (
    val id: Long,
    val address: String,
    val metropolitan: String,
    val city: String,
    val town: String,
    val road: String,
    val mainNumber: Int,
    val subNumber: Int,
    val point: Point,
){

    constructor(address: Address) : this(
        id = address.id,
        address = address.address,
        metropolitan = address.metropolitan,
        city = address.city,
        town = address.town,
        road = address.road,
        mainNumber = address.mainNumber,
        subNumber = address.subNumber,
        point = address.point
    )
}